### ISSUE TITLE
How to use extensions (curl, mysql, pgsql) with `phpenv`?

### ISSUE THREAD

**Issue by ortonomy** [phpenv/phpenv#118]
I'm sorry if I'm being dumbass, but there are no real clear instructions either with this repo or php-build how to get extensions working with this?

(I'm on arch linux) and I can install these extensions with pacman for the system version of php... but how I do I install them when I install and build php using phpenv?

**Comment by RoySegall** [comment]
No... Same as here. I'm used to `nvm` where you do `nvm install 20.0.0` and then bam! you got the version you want. This tool not work as I expected. It's installing but my global php is still what I got from brew and there are no executable in the phpenv library. 

Something here is not working smooth and I tried it for two-three times

**Comment by madumlao** [comment]
Hi.

The recommended way to add extensions is to configure your php-build environment using `PHP_BUILD_CONFIGURE_OPTS` and `PHP_BUILD_INSTALL_EXTENSION` variables. This is indicated in the README.

> By default, php-build will compile PHP with a default set of options specified by:
> 
> - php-build [default configure options](https://github.com/php-build/php-build/blob/master/share/php-build/default_configure_options)
> - per-version configure options in the PHP build definition. For example, in [7.4.13](https://github.com/php-build/php- 
    build/blob/master/share/php-build/definitions/7.4.13)
> - configure options specified in environment variables. See [the man page](https://github.com/php-build/php-build/blob/master/man/php-build.1.ronn) for details.
>
> Typically, if you need to specify how PHP is built on your system, you can add configure options in the PHP_BUILD_CONFIGURE_OPTS variable, and add PHP extensions in the PHP_BUILD_INSTALL_EXTENSION variable.

Installing php through php-build also gives you access to pecl allowing you to run `pecl install` commands and of course the modern development scene uses composer a lot. These are part of the standard php toolset.

What we probably need is documentation on the php-build side to describe how to install extensions, as it's really php-build that is doing that. There is some unwritten knowhow there that requires internal knowledge of php configure options and dependencies, including that you need to have development dependencies of each extension, since the configure script will detect which ones you have available. The most that phpenv should be doing is pointing to that info from the php-build plugin, as not all installs may be coming from php-build.

php-build options are documented in the man page, please do check

```sh
man .phpenv/plugins/php-build/man/php-build.1
```

for more info, but they do not dive into the specifics of the configure options as these vary by php version.

**Comment by madumlao** [comment]
> No... Same as here. I'm used to `nvm` where you do `nvm install 20.0.0` and then bam! you got the version you want. This tool not work as I expected. It's installing but my global php is still what I got from brew and there are no executable in the phpenv library.
> 
> Something here is not working smooth and I tried it for two-three times

@RoySegall this sounds like you have not configured your php version either globally or for your install, and is also out of scope for the question on extensions.

Also please do not expect this program to behave anything similar and the rvm family of packages (including nvm). This program follows the rbenv family of packages, which follows a very different philosophy.

**Comment by debo** [comment]
It's a good question @madumlao and I am not sure I have a good answer. As you mentioned in PHP land way before composer and other automated tools you would just pass the extra parameters to the configure script and be done with it and that's pretty much still true if we think about how `php-build` is implemented so yes, an understanding of how php works internally is beneficial and probably needed. The ask here is to create a package manager like experience really which is a whole mammoth project in its own if we want to build something universal that works across distributions and platform.

In regard to the initial question I do not thing the documentation can be made any clearer to be completely honest.
